TSResourceFile.js

```diff
@@ -0,0 +1,325 @@
+ */
+TSResourceFile.prototype.getLocale = function() {
+    return this.locale;
+};
+
+/**
+ * Get the locale of this resource file. For ts resource files, this
+ * can be extracted automatically based on the name of the directory
+ * that the file is in.
+ *
+ * @returns {String} the locale spec of this file
+ */
+TSResourceFile.prototype.getContext = function() {
+    return this.context;
+};
+
+/**
+ * Get all resources from this file. This will return all resources
+ * of mixed types (strings, arrays, or plurals).
+ *
+ * @returns {Resource} all of the resources available in this resource file.
+ */
+TSResourceFile.prototype.getAll = function() {
+    return this.set.getAll();
+};
+
+TSResourceFile.prototype.getFileName = function(pathName) {
+    if (!pathName) return;
+
+    var fileName = pathName;
+    var splitDir = fileName.split("/");
+    fileName = splitDir[splitDir.length-1];
+
+    return fileName;
+};
+
+/**
+ * Add a resource to this file. The locale of the resource
+ * should correspond to the locale of the file, and the
+ * context of the resource should match the context of
+ * the file.
+ *
+ * @param {Resource} res a resource to add to this file
+ */
+TSResourceFile.prototype.addResource = function(res) {
+    logger.trace("TSResourceFile.addResource: " + JSON.stringify(res) + " to " + this.project.getProjectId() + ", " + this.locale + ", " + JSON.stringify(this.context));
+    var resLocale = res.getTargetLocale() || res.getSourceLocale();
+    if (res && res.getProject() === this.project.getProjectId() && resLocale === this.locale.getSpec()) {
+        logger.trace("correct project, context, and locale. Adding.");
+        this.set.add(res);
+    } else {
+        if (res) {
+            if (res.getProject() !== this.project.getProjectId()) {
+                logger.warn("Attempt to add a resource to a resource file with the incorrect project.");
+            } else {
+                logger.warn("Attempt to add a resource to a resource file with the incorrect locale. " + resLocale + " vs. " + this.locale.getSpec());
+            }
+        } else {
+            logger.warn("Attempt to add an undefined resource to a resource file.");
+        }
+    }
+};
+
+/**
+ * Return true if this resource file has been modified
+ * since it was loaded from disk.
+ *
+ * @returns {boolean} true if this resource file has been
+ * modified since it was loaded
+ */
+TSResourceFile.prototype.isDirty = function() {
+    return this.set.isDirty();
+};
+
+// we don't localize resource files
+TSResourceFile.prototype.localize = function() {};
+
+function clean(str) {
+    if (!str) return;
+    return str.replace(/\s+/, " ").trim();
+}
+
+/**
+ * @private
+ */
+TSResourceFile.prototype.getDefaultSpec = function() {
+    if (!this.defaultSpec) {
+        this.defaultSpec = this.project.settings.localeDefaults ?
+            this.API.utils.getLocaleDefault(this.locale, this.flavor, this.project.settings.localeDefaults) :
```

Comment:
There is no "this.flavor" in a `TSResourceFile` right?

Reply:
Yes, `this.flavor` is always `undefined`. I've looked then I realize `getDefaultSpec` is not using anywhere. I think it's better to remove the whole function here.

Reply:
yes, please remove